modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageCompactionTriggerConfiguration.java

```diff
@@ -0,0 +1,130 @@
+
+    /**
+     * System property that defines compaction data availability time (in milliseconds).
+     *
+     * <p>Default value is {@link #DATA_AVAILABILITY_TIME_DEFAULT_VALUE}.</p>
+     */
+    public static final String DATA_AVAILABILITY_TIME_SYSTEM_PROPERTY_NAME = "metastorageCompactionDataAvailabilityTime";
+
+    /** Default value for the {@link #INTERVAL_SYSTEM_PROPERTY_NAME}. */
+    // TODO: IGNITE-23280 Make default 1 minute
+    public static final long INTERVAL_DEFAULT_VALUE = Long.MAX_VALUE;
+
+    /** Default value for the {@link #DATA_AVAILABILITY_TIME_SYSTEM_PROPERTY_NAME}. */
+    // TODO: IGNITE-23280 Make default 1 hour
+    public static final long DATA_AVAILABILITY_TIME_DEFAULT_VALUE = Long.MAX_VALUE;
+
+    private final ReadWriteLock rwLock = new ReentrantReadWriteLock();
+
+    /** Guarded by {@link #rwLock}. */
+    private long interval;
+
+    /** Guarded by {@link #rwLock}. */
+    private long dataAvailabilityTime;
+
+    /** Constructor. */
+    public MetaStorageCompactionTriggerConfiguration(SystemDistributedConfiguration systemDistributedConfiguration) {
+        updateSystemProperties(systemDistributedConfiguration.value());
+
+        systemDistributedConfiguration.listen(ctx -> {
+            updateSystemProperties(ctx.newValue());
+
+            return nullCompletedFuture();
+        });
+    }
+
+    /** Returns compaction start interval (in milliseconds). */
+    long interval() {
+        rwLock.readLock().lock();
+
+        try {
+            return interval;
+        } finally {
+            rwLock.readLock().unlock();
+        }
+    }
+
+    /** Data availability time (in milliseconds). */
+    long dataAvailabilityTime() {
+        rwLock.readLock().lock();
+
+        try {
+            return dataAvailabilityTime;
+        } finally {
+            rwLock.readLock().unlock();
+        }
+    }
+
+    private void updateSystemProperties(SystemDistributedView view) {
+        rwLock.writeLock().lock();
+
+        try {
+            interval = longValue(view, INTERVAL_SYSTEM_PROPERTY_NAME, INTERVAL_DEFAULT_VALUE);
+            dataAvailabilityTime = longValue(view, DATA_AVAILABILITY_TIME_SYSTEM_PROPERTY_NAME, DATA_AVAILABILITY_TIME_DEFAULT_VALUE);
+        } finally {
+            rwLock.writeLock().unlock();
+        }
+    }
+
+    private static long longValue(SystemDistributedView systemDistributedView, String systemPropertyName, long defaultValue) {
+        return longValue(systemDistributedView.properties().get(systemPropertyName), systemPropertyName, defaultValue);
+    }
+
+    private static long longValue(@Nullable SystemPropertyView systemPropertyView, String systemPropertyName, long defaultValue) {
+        if (systemPropertyView == null) {
+            return defaultValue;
+        }
+
+        try {
+            return Long.parseLong(systemPropertyView.propertyValue());
+        } catch (NumberFormatException e) {
+            LOG.warn(
```

Comment:
This looks dangerous. On the other hand, we cannot fail startup if there is something unparseable as this is a distributed config and we'll break the cluster. Do we have a common approach to such situations?

Reply:
It doesn't look dangerous to me. I thought about it and I think it's better not to drop the unit if the internal configuration was changed by mistake.

> Do we have a common approach to such situations?

No, this is the first such case.
I can try to discuss it with a larger group if you think it's necessary.

Reply:
If I'm not mistaken, as a result of a discussion it was decided to add validation